packages/language-server/src/lib/documents/utils.ts

```diff
@@ -11,29 +12,30 @@ export interface TagInformation {
     container: { start: number; end: number };
 }
 
-function parseAttributeValue(value: string): string {
-    return /^['"]/.test(value) ? value.slice(1, -1) : value;
-}
-
-function parseAttributes(str: string): Record<string, string> {
+function parseAttributes(attrlist: { name: string; value: string }[]): Record<string, string> {
     const attrs: Record<string, string> = {};
-    str.split(/\s+/)
-        .filter(Boolean)
-        .forEach((attr) => {
-            const [name, value] = attr.split('=');
-            attrs[name] = value ? parseAttributeValue(value) : name;
-        });
+    attrlist.forEach((attr) => {
+        attrs[attr.name] = attr.value === '' ? attr.name : attr.value; // in order to support boolean attributes (see utils.test.ts)
+    });
     return attrs;
 }
 
-const EXTRACT_TAG_EXCLUSIONS = [
-    '{#if[\\s\\S]*{\\/if}',
-    '<!--[\\s\\S]*-->',
-    '{#each[\\s\\S]*{\\/each}',
-    '{#await[\\s\\S]*{\\/await}',
-    '{@html[\\s\\S]+}',
-];
-const EXTRACT_TAG_EXCLUSION_EXPS = EXTRACT_TAG_EXCLUSIONS.map((exp) => new RegExp(exp));
+// parse5's DefaultTreeNode type is insufficient; make our own type to make TS happy
+type ParsedNode = {
+    nodeName: string;
+    tagName: string;
+    value?: string;
+    attrs: { name: string; value: string }[];
+    childNodes: ParsedNode[];
+    parentNode: ParsedNode;
+    sourceCodeLocation: Location & { startTag: Location; endTag: Location };
+};
+const re_if = new RegExp('{#if (.*?)*}', 'igms');
+const re_ifend = new RegExp('{/if}', 'igms');
+const re_each = new RegExp('{#each (.*?)*}', 'igms');
+const re_eachend = new RegExp('{/each}', 'igms');
+const re_await = new RegExp('{#await (.*?)*}', 'igms');
+const re_awaitend = new RegExp('{/await}', 'igms');
```

Comment:
placed outside to hopefully improve performance. not sure if the regex can be better here

Reply:
I replaced the whitespace with `\s`, other than that looks good.